apps/testing-javascript/src/pages/index.tsx

```diff
@@ -39,13 +42,34 @@ export const getServerSideProps: GetServerSideProps = async ({req, query}) => {
     token,
     products,
   })
+  const purchasedProductsIds =
+    commerceProps.purchases?.map((purchase) => purchase.productId) || []
+
+  const modulesAmount: {productId: string; modulesAmount: number}[] =
+    await Promise.all(
+      purchasedProductsIds.map(
+        async (
+          productId,
+        ): Promise<{productId: string; modulesAmount: number}> => {
+          const product = await getActiveProduct(productId)
+          return {productId: productId, modulesAmount: product?.modules?.length}
+        },
+      ),
+    )
+  const mostValuedProductId = modulesAmount.sort((a, b) => {
+    return b.modulesAmount - a.modulesAmount
+  })[0]?.productId
+  const mostValuedProduct = mostValuedProductId
+    ? await getActiveProduct(mostValuedProductId)
+    : null
```

Comment:
Here I define the "most valued product with maximum modules/playlists to view".  I need to define what playlists from total of 8 are available and which suppose to upgrade to get it available. I get purchased products id, get modules amount for each of them and after that define the "mostValuedProduct". Probably it's overhead and there is more smart way?